### PR TITLE
Log machine destruction and make Destroy() void

### PR DESCRIPTION
**Pull request by ajeddeloh**
Use multierror to keep trying to clean things up when destroying machines, even if part of it errors out. We should clean up as much as possible. Also log the errors in `Destroy()` itself, since the callers rarely look at the return value.

### CODE REVIEW

**Comment by ajeddeloh** [comment]
Per OOB discussion: Updated to make Destory() void in all cases and instead responsible for it's own logging. We rarely check Destoy()'s return value and even when it does return an error all we do is log it. This also makes doing `defer x.Destroy()` not silently omit errors from the logs.

**Comment by bgilbert** [comment]
Please structure the PR in such a way that every commit will compile and run.

File: platform/platform.go [hunk 1]
```diff
@@ -76,8 +77,9 @@ type Cluster interface {
 	GetDiscoveryURL(size int) (string, error)
 
 	// Destroy terminates each machine in the cluster and frees any other
-	// associated resources.
-	Destroy() error
+	// associated resources. It should log and failures; since they are not
```
Comment (bgilbert) [inline platform/platform.go]: `any`

File: platform/machine/oci/machine.go [hunk 2]
```diff
@@ -61,25 +62,21 @@ func (om *machine) Reboot() error {
 	return platform.RebootMachine(om, om.journal, om.cluster.RuntimeConf())
 }
 
-func (om *machine) Destroy() error {
-	if err := om.saveConsole(); err != nil {
-		// log error, but do not fail to terminate instance
-		plog.Error(err)
+func (om *machine) Destroy() {
+	var err multierror.Error
```
Comment (bgilbert) [inline platform/machine/oci/machine.go]: Unused.

File: platform/machine/aws/machine.go [hunk 3]
```diff
@@ -60,25 +60,21 @@ func (am *machine) Reboot() error {
 	return platform.RebootMachine(am, am.journal, am.cluster.RuntimeConf())
 }
 
-func (am *machine) Destroy() error {
+func (am *machine) Destroy() {
 	if err := am.cluster.api.TerminateInstances([]string{am.ID()}); err != nil {
-		return err
+		plog.Errorf("Error terminate instance %v: %v", am.ID(), err)
```
Comment (bgilbert) [inline platform/machine/aws/machine.go]: `terminating`

**Comment by ajeddeloh** [comment]
> Please structure the PR in such a way that every commit will compile and run.

I can, but that will squash the entire PR into one or two very large commits. I figured having more logical seperation was better, but am not too opinionated.

**Comment by bgilbert** [comment]
> I can, but that will squash the entire PR into one or two very large commits.

I think that's okay here.  IMO it's better to have one large commit that makes a simple change in many places, rather than a series of repetitive commits that are untestable.

File: platform/machine/esx/machine.go [hunk 4]
```diff
@@ -61,28 +61,24 @@ func (em *machine) Reboot() error {
 	return platform.RebootMachine(em, em.journal, em.cluster.RuntimeConf())
 }
 
-func (em *machine) Destroy() error {
+func (em *machine) Destroy() {
 	if err := em.cluster.api.TerminateDevice(em.ID()); err != nil {
-		return err
+		plog.Errorf("Error terminating instance %v: %v", em.ID(), err)
```
Comment (bgilbert) [inline platform/machine/esx/machine.go]: `terminating device`

File: platform/machine/esx/machine.go [hunk 5]
```diff
@@ -61,28 +61,24 @@ func (em *machine) Reboot() error {
...
 	if em.journal != nil {
-		if err := em.journal.Destroy(); err != nil {
-			return err
-		}
+		em.journal.Destroy()
 	}
 
 	if err := em.saveConsole(); err != nil {
-		return err
+		plog.Errorf("Error saving console for instance %v: %v", em.ID(), err)
```
Comment (bgilbert) [inline platform/machine/esx/machine.go]: `device`; same below

File: platform/machine/packet/machine.go [hunk 6]
```diff
@@ -60,19 +60,16 @@ func (pm *machine) Reboot() error {
 	return platform.RebootMachine(pm, pm.journal, pm.cluster.RuntimeConf())
 }
 
-func (pm *machine) Destroy() error {
+func (pm *machine) Destroy() {
 	if err := pm.cluster.api.DeleteDevice(pm.ID()); err != nil {
-		return err
+		plog.Errorf("Error terminating instance %v: %v", pm.ID(), err)
```
Comment (bgilbert) [inline platform/machine/packet/machine.go]: `device`

**Pull request merged**